app/assets/javascripts/listings/ListingDataService.js.coffee

```diff
@@ -0,0 +1,434 @@
+    ).cached(
+      Service.getListingsResponse(deferred, opts.retranslate)
+    ).error((data, status, headers, config) ->
+      deferred.reject(data)
+    )
+    return deferred.promise
+
+  Service.getListingsResponse = (deferred, retranslate = false) ->
+    (data, status, headers, config, itemCache) ->
+      itemCache.set(data) unless status == 'cached'
+      listings = if data and data.listings then data.listings else []
+      listings = Service.cleanListings(listings)
+      Service.groupListings(listings)
+      # On listing and listings pages, we are experiencing an issue where
+      # where the Google translation will try to keep up with digest re-calcs
+      # happening during page load and will get tripped up and fail, leaving
+      # the page untranslated. This quick fix runs the Google Translation
+      # again to cover for a possible earlier failed translate.
+      # TODO: Remove this quick fix for translation issues on listing pages
+      # and replace with a real fix based on actual digest timing.
+      $timeout(ExternalTranslateService.translatePageContent, 0, false) if retranslate
+      deferred.resolve()
+
+  Service.getListingsWithEligibility = ->
+    params =
+      householdsize: ListingEligibilityService.eligibility_filters.household_size
+      incomelevel: ListingEligibilityService.eligibilityYearlyIncome()
+      includeChildrenUnder6: ListingEligibilityService.eligibility_filters.include_children_under_6
+      childrenUnder6: ListingEligibilityService.eligibility_filters.children_under_6
+    deferred = $q.defer()
+    $http.get("/api/v1/listings/eligibility.json?#{SharedService.toQueryString(params)}",
+      { etagCache: true }
+    ).success(
+      Service.getListingsWithEligibilityResponse(deferred)
+    ).cached(
+      Service.getListingsWithEligibilityResponse(deferred)
+    ).error( (data, status, headers, config) ->
+      deferred.reject(data)
+    )
+    return deferred.promise
+
+  Service.getListingsWithEligibilityResponse = (deferred) ->
+    (data, status, headers, config, itemCache) ->
+      itemCache.set(data) unless status == 'cached'
+      listings = (if data and data.listings then data.listings else [])
+      listings = Service.cleanListings(listings)
+      Service.groupListings(listings)
+      deferred.resolve()
+
+  Service.cleanListings = (listings) ->
+    _.map listings, (listing) ->
+      # fallback for fixing the layout when a listing is missing an image
+      listing.imageURL ?= 'https://unsplash.it/g/780/438'
+    _.filter listings, (listing) ->
+      !_.includes(MAINTENANCE_LISTINGS, listing.Id)
+
+  Service.groupListings = (listings) ->
+    openListings = []
+    openMatchListings = []
+    openNotMatchListings = []
+    closedListings = []
+    lotteryResultsListings = []
+
+    listings.forEach (listing) ->
+      if ListingIdentityService.isOpen(listing)
+        # All Open Listings Array
+        openListings.push(listing)
+        if listing.Does_Match
```

Comment:
any idea what this actually does? 

Here's what the field looks like on Salesforce 
![image](https://user-images.githubusercontent.com/11825994/51708809-b06fec00-1fd9-11e9-9268-a32df9139d1b.png)

Reply:
i don't know, could you ask msol about this?